Align all 3 DK specs rune definitions

syrifgit opened the pull request:
- Nil protections were missing from Frost and Unholy `reset` function
  - Fixes #4379
- Update Blood expiry table stuff to match the newer version in UH/Frost
- Changed formatting / declaration order of model to be identical between the 3, no functional change
- Simplify some wordy comments, removed duplicated ones, whitespace stuff, etc.

syrifgit committed:
Align all 3 DK specs rune definitions

Nil protections were missing from Frost and Unholy. Changed formatting to be identical between the 3. Simplify some wordy comments.

bjth commented on TheWarWithin/DeathKnightBlood.lua:
```diff
@@ -47,51 +47,45 @@ spec:RegisterResource( Enum.PowerType.Runes, {
         table.sort( t.expiry )
-
-        t.actual = nil
+        t.actual = nil -- Reset actual to force recalculation
```
Comment got added here, but taken out of the other places, want to be consistent and remove it?

syrifgit replied:
Imagine being consistent, couldn't be me

syrifgit committed:
Cleanup comments

ty Mondo, I intended to leave the comment once on each spec under `reset` as a reminder.
